lib/types/documentarray.js

```diff
@@ -180,6 +180,11 @@ class CoreDocumentArray extends CoreMongooseArray {
     return arr;
   }
 
+  map() {
+    const arr = [].concat(Array.prototype.map.apply(this,arguments));
```

Comment:
I think this change is currently unnecessary. I like the idea of `map()`, `filter()`, etc. returning vanilla JavaScript arrays rather than mongoose arrays, but I think that's a bit too heavy of a change for a patch release. I think that all we need for this release is the `if (!parent || parent.$__.populated == null) return;` change.